libs/ledger/src/upow/synergetic_executor.cpp

```diff
@@ -87,15 +111,34 @@ void SynergeticExecutor::Verify(WorkQueue &solutions, ProblemData const &problem
       contract->Attach(storage_);
       ContractContext ctx(&token_contract_, solution->address(), &storage_adapter, 0);
       contract->UpdateContractContext(ctx);
-      status = contract->Complete(solution->address(), shard_mask);
-      contract->Detach();
+
+      // TODO(AB): charge limit
+      FeeManager::TransactionDetails tx_details{
+          solution->address(), solution->address(), shard_mask, solution->contract_digest(),
+          CHARGE_RATE,         CHARGE_LIMIT,        false};
```

Comment:
This is punishing for the solution provider if the complete part of the contract turns out to be error prone